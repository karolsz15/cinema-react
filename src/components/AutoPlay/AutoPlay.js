import React, { Component } from "react";
import Slider from "react-slick";

class AutoPlay extends Component {
  render() {

    const settings = {
      dots: false,
      infinite: true,
      slidesToShow: 3,
      slidesToScroll: 1,
      autoplay: true,
      speed: 2000,
      autoplaySpeed: 2000,
      cssEase: "linear"
    };
    
    return (
      <div style={{padding: '1em'}}>
        <Slider {...settings}>
          <div>
            <img src={this.props.poster1} alt="" />
          </div>
          <div>
            <img src={this.props.poster2} alt="" />
          </div>
          <div>
            <img src={this.props.poster3} alt="" />
          </div>
          <div>
            <img src={this.props.poster4} alt="" />
          </div>
          <div>
            <img src={this.props.poster5} alt="" />
          </div>
          <div>
            <img src={this.props.poster6} alt="" />
          </div>
          <div>
            <img src={this.props.poster7} alt="" />
          </div>
          <div>
            <img src={this.props.poster8} alt="" />
          </div>
        </Slider>
      </div>
    );
  }
};

export default AutoPlay;