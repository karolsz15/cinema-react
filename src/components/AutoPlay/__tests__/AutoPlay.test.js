import React from 'react';
import { shallow } from 'enzyme';
import AutoPlay from '../AutoPlay';

const wrapper = shallow(<AutoPlay />);

describe('AutoPlay component', () => {
  it('should render the AutoPlay correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
