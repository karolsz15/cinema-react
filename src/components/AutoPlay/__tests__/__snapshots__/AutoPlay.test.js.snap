// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoPlay component should render the AutoPlay correctly 1`] = `
<div
  className="sliderContainer"
  style={
    Object {
      "padding": "1em",
    }
  }
>
  <Slider
    autoplay={true}
    autoplaySpeed={2000}
    cssEase="linear"
    dots={false}
    infinite={true}
    slidesToScroll={1}
    slidesToShow={4}
    speed={2000}
  />
</div>
`;
